### ISSUE TITLE
Fix `amplitude_to_DB` numerical instability

### ISSUE THREAD

**Issue by mthrok** [pytorch/audio#611]
Recently we discovered that `torch.log10` behaves differently on different CPUs when the input tensor is float32. https://github.com/pytorch/pytorch/issues/37833 **Note**: This is not a random behavior.

Due to this, we have some tests that passes on some machines but fail on other machines.
A proposed fix for this is to use `float64` type before passing the tensor to `torch.log10` [here](https://github.com/pytorch/audio/blob/b71b5ee56c7be8b7cecd58b7d44b2707c1e4f5af/torchaudio/functional.py#L395), and convert back to `float32` if the input type was `float32`

Once this is completed, we could re-enable the following tests;

1. [`librosa_compatibility_test`](https://github.com/pytorch/audio/blob/b71b5ee56c7be8b7cecd58b7d44b2707c1e4f5af/test/test_librosa_compatibility.py#L247)
 2. [`test_batch_consistency`](https://github.com/pytorch/audio/blob/master/test/test_batch_consistency.py#L210-L219)

For 2. See the commit b499840 and revert the `atol` to `1e-5`.

You can run these test with the following commands;
```
pytest test -v -k batch_mfcc
pytest test -v -k basics3
```

In addition to that, all the existing test that uses `amplitude_to_DB`, should pass.
```
pytest test -v -k amplitude_to_DB
```

**Comment by Micheladar** [comment]
I'll work on it.

**Comment by kshitij12345** [comment]
@Micheladar Hi, are you working on it?

**Comment by discort** [comment]
@mthrok 
Is still relevant issue? If so, could you please update the description. It seems there are no mentioned tests

**Comment by mthrok** [comment]
Resolved via #1436 #1439

@kshitij12345 @discort sorry for not responding earlier. The test has been refactored and refined. This issue is no longer relevant.
